sources/packages/backend/apps/api/src/services/cas-invoice-batch/cas-invoice-batch.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { CASInvoiceBatch } from "@sims/sims-db";
+import { In, Repository } from "typeorm";
+import {
+  CASInvoiceBatchesPaginationOptions,
+  PaginatedResults,
+} from "../../utilities";
+
+@Injectable()
+export class CASInvoiceBatchService {
+  constructor(
+    @InjectRepository(CASInvoiceBatch)
+    private readonly casInvoiceBatchRepo: Repository<CASInvoiceBatch>,
+  ) {}
+
+  /**
+   * Retrieve all CAS invoice batches.
+   * @param paginationOptions pagination options.
+   * @returns paginated CAS invoice batches results.
+   */
+  async getCASInvoiceBatches(
+    paginationOptions: CASInvoiceBatchesPaginationOptions,
+  ): Promise<PaginatedResults<CASInvoiceBatch>> {
+    const [results, count] = await this.casInvoiceBatchRepo.findAndCount({
+      select: {
+        id: true,
+        batchName: true,
+        batchDate: true,
+        approvalStatus: true,
+        approvalStatusUpdatedOn: true,
+        approvalStatusUpdatedBy: { id: true, firstName: true, lastName: true },
+      },
+      relations: {
+        approvalStatusUpdatedBy: true,
+      },
+      where: {
+        approvalStatus: paginationOptions.approvalStatusSearch?.length
+          ? In(paginationOptions.approvalStatusSearch)
+          : undefined,
+      },
+      skip: paginationOptions.pageLimit * paginationOptions.page,
+      take: paginationOptions.pageLimit,
+      order: {
+        [paginationOptions.sortField]: paginationOptions.sortOrder,
```

Comment:
Can we have a default sort-field and order from backend perspective? or Are we expecting the Vue to handle the default sort field and order?

Reply:
From the moment that we exposed the option to the API to receive the sort, I did not believe that required a default.